{
    "url": "https://api.github.com/repos/dask/dask/issues/3976",
    "repository_url": "https://api.github.com/repos/dask/dask",
    "labels_url": "https://api.github.com/repos/dask/dask/issues/3976/labels{/name}",
    "comments_url": "https://api.github.com/repos/dask/dask/issues/3976/comments",
    "events_url": "https://api.github.com/repos/dask/dask/issues/3976/events",
    "html_url": "https://github.com/dask/dask/issues/3976",
    "id": 359336553,
    "node_id": "MDU6SXNzdWUzNTkzMzY1NTM=",
    "number": 3976,
    "title": "Inconsistent behaviour when calculating sizeof GeoDataFrame and GeoSeries",
    "user": {
        "login": "avnovikov",
        "id": 15693459,
        "node_id": "MDQ6VXNlcjE1NjkzNDU5",
        "avatar_url": "https://avatars.githubusercontent.com/u/15693459?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/avnovikov",
        "html_url": "https://github.com/avnovikov",
        "followers_url": "https://api.github.com/users/avnovikov/followers",
        "following_url": "https://api.github.com/users/avnovikov/following{/other_user}",
        "gists_url": "https://api.github.com/users/avnovikov/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/avnovikov/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/avnovikov/subscriptions",
        "organizations_url": "https://api.github.com/users/avnovikov/orgs",
        "repos_url": "https://api.github.com/users/avnovikov/repos",
        "events_url": "https://api.github.com/users/avnovikov/events{/privacy}",
        "received_events_url": "https://api.github.com/users/avnovikov/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [],
    "state": "open",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 5,
    "created_at": "2018-09-12T06:43:28Z",
    "updated_at": "2018-10-22T21:09:52Z",
    "closed_at": null,
    "author_association": "NONE",
    "active_lock_reason": null,
    "body": "As found in discussion in \r\nhttps://github.com/dask/dask/issues/3972#issuecomment-420384065 and probably the root cause of https://github.com/dask/dask/issues/3972\r\n\r\n```\r\nAttributeError: 'GeometryArray' object has no attribute 'nbytes'\r\ndistributed.sizeof - WARNING - Sizeof calculation failed.  Defaulting to 1MB\r\nTraceback (most recent call last):\r\n  File \"/anaconda3/envs/mariquant/lib/python3.6/site-packages/distributed/sizeof.py\", line 16, in safe_sizeof\r\n    return sizeof(obj)\r\n  File \"/anaconda3/envs/mariquant/lib/python3.6/site-packages/dask/utils.py\", line 415, in __call__\r\n    return meth(arg)\r\n  File \"/anaconda3/envs/mariquant/lib/python3.6/site-packages/dask/sizeof.py\", line 63, in sizeof_pandas_series\r\n    p = int(s.memory_usage(index=True))\r\n  File \"/anaconda3/envs/mariquant/lib/python3.6/site-packages/pandas/core/series.py\", line 3503, in memory_usage\r\n    v = super(Series, self).memory_usage(deep=deep)\r\n  File \"/anaconda3/envs/mariquant/lib/python3.6/site-packages/pandas/core/base.py\", line 1143, in memory_usage\r\n    v = self.values.nbytes\r\n```\r\n\r\nHowever GeoDataFrame size seems to be calculated correctly. Looking at the code of distributed/sizeof.py the only suspicious line is `p = int(s.memory_usage(index=True))`, however\r\n\r\n```\r\ntype(anchorage_buffer)\r\nOut[55]:\r\ngeopandas.geoseries.GeoSeries\r\nIn [57]:\r\nanchorage_buffer.memory_usage(index=True)\r\nI am densified (external_values, 20353 elements)\r\nI am densified (external_values, 20353 elements)\r\nOut[57]:\r\n325648\r\n```\r\nUnfortunately I can't trace the origin of this error.\r\n\r\nThis issue seems to be most GeoPandas related and as such I posted a companion issue [#819](https://github.com/geopandas/geopandas/issues/819)",
    "closed_by": null,
    "reactions": {
        "url": "https://api.github.com/repos/dask/dask/issues/3976/reactions",
        "total_count": 0,
        "+1": 0,
        "-1": 0,
        "laugh": 0,
        "hooray": 0,
        "confused": 0,
        "heart": 0,
        "rocket": 0,
        "eyes": 0
    },
    "timeline_url": "https://api.github.com/repos/dask/dask/issues/3976/timeline",
    "performed_via_github_app": null,
    "state_reason": null
}